packages/app-builder-lib/src/util/AppFileWalker.ts

```diff
@@ -62,17 +76,30 @@ function createAppFilter(matcher: FileMatcher, packager: Packager): Filter | nul
 
   const filter = matcher.createFilter()
   return (file, fileStat) => {
+    const matchesFilter = filter(file, fileStat)
     if (!nodeModulesFilter(file, fileStat)) {
-      return false
+      //  it's a node_modules directory
+
+      //  if includeSubNodeModules is true, then we just return true - we want
+      //  all of them
+      if (includeSubNodeModules) {
+        return true
+      }
     }
-    return filter(file, fileStat)
+    return matchesFilter
   }
 }
 
 /** @internal */
 export class AppFileWalker extends FileCopyHelper implements FileConsumer {
+  readonly includeSubNodeModules: boolean
+  readonly matcherFilter
+
   constructor(matcher: FileMatcher, packager: Packager) {
     super(addAllPatternIfNeed(matcher), createAppFilter(matcher, packager), packager)
+    //  @ts-ignore
+    this.includeSubNodeModules = (packager._configuration || {}).includeSubNodeModules;
```

Comment:
Another instance of a ts-ignore that could be fixed by better typing.

At the very least a type cast would be better.

Reply:
See above.